server/plugin.go

```diff
@@ -45,8 +52,35 @@ type Plugin struct {
 
 func (p *Plugin) OnActivate() error {
 	p.logger = &metricsLogger{api: p.API}
-
 	appCfg := p.API.GetConfig()
+
+	p.closeChan = make(chan bool)
+	p.waitGroup = sync.WaitGroup{}
+	
+	// we are using a mutually exclusive lock to run a single instance of this plugin
+	// we don't really need to collect metrics twice, in any case we can still collect
+	// twice, TSDB will takce care of overlapped blocks. But it will increase the disk
+	// writes to the remote or local disk.
```

Comment:
Nit: typo and rewording
```suggestion
	// we are using a mutually exclusive lock to run a single instance of this plugin
	// we don't really need to collect metrics twice: although TSDB will take care
	// of overlapped blocks, it will increase the disk writes to the remote or local
	// disk.
```